jaeger-zipkin/src/main/java/io/jaegertracing/senders/zipkin/V2SpanConverter.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2018, The Jaeger Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.jaegertracing.senders.zipkin;
+
+import com.google.gson.Gson;
+import io.jaegertracing.Constants;
+import io.jaegertracing.LogData;
+import io.jaegertracing.Span;
+import io.jaegertracing.SpanContext;
+import io.jaegertracing.Tracer;
+import io.opentracing.tag.Tags;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import zipkin2.Endpoint;
+
+/**
+ * Converts a Jaeger span to a Zipkin2 span.
+ */
+public class V2SpanConverter {
+
+  private static final Gson gson = new Gson();
+
+  public static zipkin2.Span convertSpan(Span span) {
+    Tracer tracer = span.getTracer();
+    Endpoint host = zipkin2.Endpoint.newBuilder()
+        .ip(convertIp(tracer.getIpv4()))
+        .port(0)
+        .serviceName(tracer.getServiceName())
+        .build();
+
+    Endpoint peerEndpoint = extractPeerEndpoint(span.getTags());
+
+    SpanContext context = span.context();
+    zipkin2.Span.Builder builder = zipkin2.Span.newBuilder()
+            .id(Long.toHexString(context.getSpanId()))
+            .traceId(Long.toHexString(context.getTraceId()))
+            .name(span.getOperationName())
+            .parentId(Long.toHexString(context.getParentId()))
+            .debug(context.isDebug())
+            .localEndpoint(host)
+            .remoteEndpoint(peerEndpoint)
+            .kind(convertKind(span.getTags().get(Tags.SPAN_KIND.getKey())))
+            .timestamp(span.getStart())
+            .duration(span.getDuration());
+
+    buildAnnotations(span, builder);
+    buildTags(span, builder);
+
+    return builder.build();
+  }
+
+  private static zipkin2.Span.Kind convertKind(Object kind) {
+    if (kind == null) {
```

Comment:
If you reverse your equals calls you don't need that initial null check and have a single location where you return null from this method